rllib/agents/trainer.py

```diff
@@ -1078,6 +1055,18 @@ def make_callbacks():
 
     @staticmethod
     def _validate_config(config: PartialTrainerConfigDict):
+        # Resolve `num_gpus`=None (default): Set `num_gpus` to as many as we
```

Comment:
This is the crux of this PR: How do we figure out, how many GPUs are available if `num_gpus: None`?